lib/gollum/app.rb

```diff
@@ -435,12 +436,12 @@ class App < Sinatra::Base
       end
 
       post '/compare/*' do
-        @file     = encodeURIComponent(params[:splat].first)
+        @file     = encodeURIComponent(params[:splat].first).gsub('%2F', '/')
```

Comment:
This will mean the `%2F` will show up on the page if `@file` is used in the template -- that's why it's arguably better to take care of 'cleaning up' the URL at the point of the the redirect call. Maybe just leave that out completely for now, and figure out how to clean up the URLs in a separate PR where we tackle the issue of which encoding method to use?